securedrop_proxy/entrypoint.py

```diff
@@ -14,76 +14,68 @@
 
 from logging.handlers import TimedRotatingFileHandler
 
-from securedrop_proxy import callbacks
-from securedrop_proxy import config
 from securedrop_proxy import main
 from securedrop_proxy import proxy
 from securedrop_proxy.version import version
 
 
 DEFAULT_HOME = os.path.join(os.path.expanduser("~"), ".securedrop_proxy")
-LOGLEVEL = os.environ.get('LOGLEVEL', 'info').upper()
+LOGLEVEL = os.environ.get("LOGLEVEL", "info").upper()
 
 
-def start():
-    '''
+def start() -> None:
+    """
     Set up a new proxy object with an error handler, configuration that we read from  argv[1], and
     the original user request from STDIN.
-    '''
+    """
     try:
         configure_logging()
 
-        logging.debug('Starting SecureDrop Proxy {}'.format(version))
-
-        # a fresh, new proxy object
-        p = proxy.Proxy()
-
-        # set up an error handler early, so we can use it during
-        # configuration, etc
-        original_on_done = p.on_done
-        p.on_done = callbacks.err_on_done
+        logging.debug("Starting SecureDrop Proxy {}".format(version))
 
         # path to config file must be at argv[1]
         if len(sys.argv) != 2:
-            raise ValueError("sd-proxy script not called with path to configuration file")
+            raise ValueError(
+                "sd-proxy script not called with path to configuration file"
+            )
 
-        # read config. `read_conf` will call `p.on_done` if there is a config
+        # read config. `read_conf` will call `p.err_on_done` if there is a config
         # problem, and will return a Conf object on success.
         conf_path = sys.argv[1]
-        p.conf = config.read_conf(conf_path, p)
+        # a fresh, new proxy object
+        p = proxy.Proxy(conf_path=conf_path)
 
         # read user request from STDIN
-        incoming = []
```

Comment:
I think this is because the examples has new lines in it, and when josh started to write the tool, having the newlines and formatted text helped to debug. I kept it the same, as if in future some other part of the code sends in large part of text with new lines (via a few write calls in the other side of `qrexec` service), also I am not very sure about how qrexec service writes those lines. That is why I kept it the same.

Reply:
OK. Let's investigate separately.